src/main/java/ru/testtask/service/FileService.java

```diff
@@ -0,0 +1,250 @@
+        try {
+            URL url = new URL(uploadFileDTO.getFileUrl());
+            String filename = UUID.randomUUID().toString() + "." + FilenameUtils.getExtension(url.toString());
+            temporaryDir = Files.createTempDirectory(storageRootPath, "temp");
+            Path path = storageRootPath.resolve(filename);
+            File tempFile = uploadFileFromUrl(url);
+
+            FileData fileData = FileData.builder().originalFilename(uploadFileDTO.getFileName())
+                    .filename(filename)
+                    .size(Files.size(tempFile.toPath()))
+                    .type(Files.probeContentType(path))
+                    .directory("/")
+                    .build();
+
+            fileDataRepo.insert(fileData);
+            Files.move(tempFile.toPath(), path);
+            Files.deleteIfExists(tempFile.toPath());
+            Files.deleteIfExists(temporaryDir);
+            return fileData;
+        } catch (IOException e) {
+            log.error("An error occurred during uploading file");
+            return null;
+        }
+    }
+
+    public List<FileData> getFileListFromDirectory(String directory){
+        String regex = "^/" + directory;
+        return fileDataRepo.findFileDataByRegexpDirectory(regex);
+    }
+
+    public void removeFile(String id) throws FileNotFoundException {
+        Optional<FileData> optionalFileData = fileDataRepo.findById(id);
+            if (optionalFileData.isPresent()){
+                try {
+                    FileData fileData = optionalFileData.get();
+                    Files.deleteIfExists(getFileAbsolutePath(fileData));
+                    fileDataRepo.deleteById(id);
+                } catch (IOException e){
+                    log.error("Impossible to remove file", e);
+                }
+            }
+            else {
+                throw new FileNotFoundException(String.format("File with id %s does not found", id));
+            }
+        }
+
+    public List<FileData> searchByRegex(String regex){
+        return fileDataRepo.findFileDataByRegexpFilename(regex);
+    }
+
+    public FileData moveFile(String id, String directory) throws FileNotFoundException {
+        Optional<FileData> optionalFileData = fileDataRepo.findById(id);
+        if (optionalFileData.isEmpty()){
+            throw new FileNotFoundException(String.format("File with id %s does not found", id));
+        }
+            FileData fileData = optionalFileData.get();
+            Path destDirectory = storageRootPath.resolve(directory);
+            Path destination = destDirectory.resolve(fileData.getFilename());
+
+        try {
+            if (!Files.exists(destDirectory)) {
+                Files.createDirectories(destDirectory);
+            }
+
+            Files.move(getFileAbsolutePath(fileData), destination);
+            fileData.setDirectory("/" + directory);
+            fileDataRepo.save(fileData);
```

Comment:
It is better to try to save data to db before the file moving. 
And you need to add `finally` block to rollback any changes if some step is failed.